cmd/influxd/run/command.go

```diff
@@ -65,27 +65,38 @@ func NewCommand() *Command {
 	}
 }
 
-// Run parses the config from args and runs the server.
-func (cmd *Command) Run(args ...string) error {
+func (cmd *Command) LoadConfig(args ...string) (Options, *Config, error) {
+	fail := func(err error) (Options, *Config, error) { return Options{}, nil, err }
+
 	// Parse the command line flags.
 	options, err := cmd.ParseFlags(args...)
 	if err != nil {
-		return err
+		return fail(fmt.Errorf("error parsing command line: %w", err))
 	}
 
 	config, err := cmd.ParseConfig(options.GetConfigPath())
 	if err != nil {
-		return fmt.Errorf("parse config: %s", err)
+		return fail(fmt.Errorf("error parsing config file: %s", err))
```

Comment:
Add the config file path here so we can figure out bad environment variables, etc.